### ISSUE TITLE
Can't enable lib checking in consumer tsconfig.json with ESM

### ISSUE THREAD

**Issue by fubhy** [connectrpc/connect-es#679]
**Describe the bug**

Currently, types are generated using ESM config but, unlike the files in `dist/esm`, do not have a parent `package.json` file with `type: module`. Hence, the types are considered CJS and cause typechecking errors when a consumer enables lib checking in their `tsconfig.json`.

**To Reproduce**

Set `skipLibCheck: false` in a consuming package and import from `@bufbuild/connect`. Since that imports from `@bufbuild/protobuf` (which has the types exported as ESM), you'll get the following error (and many others like it):

```
node_modules/@bufbuild/connect/dist/types/router.d.ts(1,46): error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("@bufbuild/protobuf")' call instead.
```

**Comment by timostamm** [comment]
Just catching up 😅 

Good call - sounds like the "Masquerading as CJS" issue reported by arethetypeswrong. This should be fixed through https://github.com/connectrpc/connect-es/issues/801.

**Comment by smaye81** [comment]
Hey @fubhy can your share your tsconfig `compilerOptions`? I'm guessing this is with `moduleResolution` of `Node16` correct?

**Comment by smaye81** [comment]
This should be fixed in the v1.1.0 release.  Release notes [here](https://github.com/connectrpc/connect-es/releases/tag/v1.1.0).
